File: dsp02.c
#include <graph.h>
#include <conio.h>
#include <stdio.h>
#include <math.h>

#define XDIM 100

float   fX[XDIM + 1], fB = 0.0, fW = 0.0;

void main (void)
    {
    /*
     ***  PROGRAM NO.2  SAMPLED EXPONENTIALS AND SINUSOIDS  ****
     */
    int    iCount;

    _clearscreen  (_GCLEARSCREEN);
    printf ("PROGRAM 2: Sampled exponentials and sinusoids\n");

    printf ("Enter value of Beta-Zero :");
    scanf ("%f", &fB);

    printf ("Enter value of Omega :");
    scanf ("%f", &fW);

    for (iCount = 1; iCount <= XDIM; iCount++)
        fX[iCount] = exp (fB * (float)iCount) *
                     cos (fW * (float)iCount);

    /*
     ************** DRAW AXES, THEN SIGNAL SAMPLES *************
     */
    _setvideomode( _MRES16COLOR );

    _moveto (1, 100);   _lineto (300, 100);

    for (iCount = 1; iCount <= XDIM; iCount++)
        {
        _moveto (3 * iCount, 100);
        _lineto (3 * iCount, (int)(100.0 - fX[iCount] * 30.0 + 0.5));
        }

    getch();
    _setvideomode( _DEFAULTMODE );

    /*
     ***********************************************************
     */
    }

File: dsp02.pas
(*****  PROGRAM NO.2    SAMPLED EXPONENTIALS AND SINUSOIDS  ****)
program DSP02;
uses graph3;
var X: array[1..100] of real;
var N: integer;
var B,W: real;
begin
  writeln('Value of beta-zero?');
  readln(B);
  writeln('Value of omega?');
  readln(W);
    begin
     for N:=1 to 100 do
       begin
         X[N]:=exp(B*N)*cos(W*N);
       end;
(**************  DRAW AXES, THEN SIGNAL SAMPLES  ***************)
     graphmode;
     draw(1,100,300,100,1);
     draw(1,0,1,200,1);
     for N:=1 to 100 do
       begin
         draw(3*N,100,3*N,round(100-X[N]*30),1);
       end;
    end;
end.
(***************************************************************)